boards/arm/nrf52_sparkfun/nrf52_sparkfun.dts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf52832.dtsi>
+
+/ {
+	model = "Sparkfun nRF52832 Breakout";
+	compatible = "sparkfun,nrf52832", "nordic,nrf52832-qfaa",
+		     "nordic,nrf52832";
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	status = "ok";
+};
+
+&flash0 {
+	/*
+	 * If chosen's zephyr,code-partition
+	 * is unset, the image will be linked
+	 * into the entire flash device.  If
+	 * it points to an individual
+	 * partition, the code will be linked
+	 * to, and restricted to that
+	 * partition.
+	 */
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x8000>;
```

Comment:
in sucha a small boot partition mcuboot with serial recovery enabled will not fit.  We use 0xc000 bouns for nrf52_dk for instance. This settings has no impact on either non-bootable or no-bootloader builds (so default one) .Did you considered this?

Reply:
I've only used this code in the no-bootloader case.

The interaction between mcuboot and zephyr seems completly undocumented. I'd have considered it if I had the least idea where to start. Suggestions welcome. 

I have a project that requires mcuboot in the pipe, but for now this is a directly copy of the blenano2 overlay.

Reply:
I ended up copying over the pca10040 flash layout. I didn't realize that BLENANO2 was out-of-date.